src/Tools/BuildValidator/LocalReferenceResolver.cs

```diff
@@ -99,21 +76,20 @@ public void CacheNames(ImmutableArray<MetadataReferenceInfo> names)
                         continue;
                     }
 
-                    var mvids = GetMvidsForFile(file);
+                    var mvid = GetMvidForFile(file);
+                    if (!mvid.HasValue || _cache.ContainsKey(mvid.Value))
+                    {
+                        continue;
+                    }
 
-                    foreach (var match in potentialMatches)
+                    var matchedReference = potentialMatches.SingleOrDefault(m => m.Mvid == mvid);
```

Comment:
```suggestion
                    var matchedReference = potentialMatches.FirstOrDefault(m => m.Mvid == mvid);
```

A bit of an edge case but `SingleOrDefault` would file in the case where a package was *evil* and decided to have files differing by only case in the same directory. This requires a bit of evilness on the package author part. At the same time though this tool is doing a fairly broad search on the machine for PE files. Would avoid crashing if that occurs and just move on.